force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +145,38 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        // Get or create Account that has the name accountName.
+        Account acc = new Account();
+        List<Account> accounts = [SELECT Id, Name FROM Account WHERE Name = :accountName];
+        if (accounts.size() == 0) {
+            acc.Name = accountName;
+            insert acc;
+        } else {
+            acc = accounts.get(0);
+        }
+        // Get all Opportunities associated with the Account.
+        List<Opportunity> opportunities = [SELECT Id, Name FROM Opportunity WHERE AccountId = :acc.Id];
+        // Create new opportunities with provided names
+        if (opportunities.size() == 0) {
+            for (String oppName : oppNames) {
+                Opportunity opp = new Opportunity(Name = oppName, AccountId = acc.Id);
+                opp.StageName = 'Qualification';
+                opp.CloseDate = Date.today().addMonths(3);
+                opportunities.add(opp);
+            }
+        } else {
+            // Check all names and update if they are not in the list. 
+            // Assume: we have the same amount of records as the number of provided names
+            if (opportunities.size() == oppNames.size()) {
+                for (Opportunity opp : opportunities) {
+                    if (!oppNames.contains(opp.Name)) {
+                        Integer i = opportunities.indexOf(opp);
+                        opp.Name = oppNames.get(i);
+                    }
+                }
+            }
+        }
```

Comment:
this is about as good as it gets without using a map for the else block. You can use maps to figure out which you do and don't have based on the opportunity Name